### PR TITLE
Add parameter to configure heap size

### PR DESCRIPTION
**Pull request by rishabh6788**
### Description
This PR adds capability to set JVM heap size of the OS cluster. By default the min and max heap size set is 1G in `jvm.options` file.

### Issues Resolved
_List any issues this PR will resolve, e.g. Closes [...]._ 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -58,6 +58,8 @@ In order to deploy both the stacks the user needs to provide a set of required a
...
 | dataNodeStorage (Optional)        | string  | User provided ebs block storage size, defaults to 100Gb                                                                                                                                                                                                                                          |
 | mlNodeStorage (Optional)          | string  | User provided ebs block storage size, defaults to 100Gb                                                                                                                                                                                                                                          |
+| heapSize (Optional)               | string  | User provided JVM heap memory size (integer) in Gb, defaults to 1Gb, e.g. `--context heapSize=2` to set heap to 2Gb                                                                                                                                                                              |
```
Comment (gaiksaya) [inline README.md:61]: I think better to keep the naming same as property `jvmHeapSize`. Also shouldn't this be an integer?
```suggestion
| jvmHeapSize (Optional)               | integer  | User provided JVM heap memory size (integer) in Gb, defaults to 1Gb, e.g. `--context heapSize=2` to set heap to 2Gb                                                                                                                                                                              |
```

File: lib/infra/infra-stack.ts [hunk 2]
```diff
@@ -55,6 +55,7 @@ export interface infraProps extends StackProps{
     readonly mlNodeStorage: number,
     readonly jvmSysPropsString?: string,
     readonly additionalConfig?: string,
+    readonly jvmHeapSize?: number,
```
Comment (gaiksaya) [inline lib/infra/infra-stack.ts:58]: It's an integer here so should be the same above

**Review by gaiksaya — APPROVED** [review]
LGTM! 
Please squash multiple commits into 1 clean commit before merging.
Thanks!

File: README.md [hunk 3]
```diff
@@ -58,6 +58,8 @@ In order to deploy both the stacks the user needs to provide a set of required a
...
 | dataNodeStorage (Optional)        | string  | User provided ebs block storage size, defaults to 100Gb                                                                                                                                                                                                                                          |
 | mlNodeStorage (Optional)          | string  | User provided ebs block storage size, defaults to 100Gb                                                                                                                                                                                                                                          |
+| jvmHeapSize (Optional)            | integer | User provided JVM heap memory size (integer) in Gb, defaults to 1Gb, e.g. `--context jvmHeapSize=2` to set heap to 2Gb                                                                                                                                                                           |
```
Comment (dreamer-89) [inline README.md:61]: nit: Is it possible to differentiate b/w data and cluster manager nodes. This setting may not be needed for all type of nodes.

**Pull request merged**